lib/weather_cast_angle/services/tide_data_handler.ex

```diff
@@ -2,31 +2,71 @@ defmodule WeatherCastAngle.Services.TideDataHandler do
   @target_url "https://www.data.jma.go.jp/gmd/kaiyou/data/db/tide/suisan/txt/"
 
   @doc """
-  HTTP GET request and and return response body.
+  Get tide data map by HTTP requested result or cached value.
+
+  ## Parameters
+    - `year`: The input year as an example, it is a 4-digit integer, such as 2024 or 2025.
+    - `location_code`: The input Location Code in String format.
+
+  ## Returns
+    - A map where:
+      - The keys are date strings (e.g., "2024-01-01"). Equals to `target_date`.
+      - The values are maps with the following structure:
+        - `"hourly_tide_levels"`: A list of integers representing the tide levels for each hour.
+        - `"target_date"`: A string representing the date for the tide data.
+        - `"location_code"`: A string code representing the location of the tide measurement.
+        - `"high_tide"`: A list of maps, each with a string key representing a time and an integer value representing the tide level at that time.
+        - `"low_tide"`: Similar to `"high_tide"`, but for low tide measurements.
+    - If the cache does NOT exist and an HTTP request GET error occurs,
+      `%{"Error" => reason}` formatted map is returned.
   """
-  @spec get_response_body(pos_integer, String.t()) :: %{
-          String.t() => %{
-            hourly_tide_levels: [integer()],
-            target_date: String.t(),
-            location_code: String.t(),
-            high_tide: [{String.t(), integer()}],
-            low_tide: [{String.t(), integer()}]
+  @spec get_tide_data(pos_integer, String.t()) ::
+          %{
+            String.t() => %{
+              String.t() => [integer()],
+              String.t() => String.t(),
+              String.t() => String.t(),
+              String.t() => [%{String.t() => integer()}],
+              String.t() => [%{String.t() => integer()}]
+            }
```

Comment:
[comment] Really want to make the key in value map explicit by giving concrete string information, but since this is not possible in use, so use `String.t()` all.